File: quiz/src/main/java/tn/esprit/entities/Quiz.java
package tn.esprit.entities;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.io.Serializable;

@Entity
@Table(name = "quizzes")
@Setter
@Getter
@NoArgsConstructor
@AllArgsConstructor
public class Quiz implements Serializable {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private String title;
    private String description;
    private Integer duration;
}

File: src/app/model/Quiz.ts
export class Quiz {
  id: number;
  title: string;
  description: string;
  duration: number;

  constructor(id: number, title: string, description: string, duration: number) {
    this.id = id;
    this.title = title;
    this.description = description;
    this.duration = duration;
  }
}
